tee-worker/ts-tests/common/utils.ts

```diff
@@ -282,24 +282,25 @@ export async function checkJSON(vc: any, proofJson: any): Promise<boolean> {
     expect(isValid).to.be.true;
     expect(
         vc.type[0] === 'VerifiableCredential' &&
-        vc.issuer.id === proofJson.verificationMethod &&
-        proofJson.type === 'Ed25519Signature2020'
+            vc.issuer.id === proofJson.verificationMethod &&
+            proofJson.type === 'Ed25519Signature2020'
     ).to.be.true;
     return true;
 }
 
-export async function checkFailReason(
+export async function checkErrorDetail(
     response: string[] | Event[],
-    expectedReason: string,
+    expectedDetail: string,
     isModule: boolean
 ): Promise<boolean> {
-    let failReason = '';
+    let detail: string = '';
+    // TODO: sometimes `item.data.detail.toHuman()` or `item` is treated as object (why?)
```

Comment:
JSON.stringify() cannot guarantee to return a string type, it will return an `object` such as `{func:function(){}}`. This might be one of the reasons why you are seeing the error,  can console the `removeErrorIdentities events[i].data.detail.toHuman() `type to confirm,may be it is a function type.